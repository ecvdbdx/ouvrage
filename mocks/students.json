[
  {
    "username": "epatrois",
    "first_name": "Manu",
    "last_name": "Patrois",
    "id": "2"
  },
  {
    "id": "5",
    "username": "mdewaele",
    "first_name": "Maxence",
    "last_name": "De Waele",
    "email": "maxencedewaele@gmail.com",
    "birthdate": "838921463",
    "avatar": "https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png"

  },
  {
    "id": "8",
    "username": "rodrigodelanoche",
    "first_name": "Vincent",
    "last_name": "Rodriguez",
    "email": "vincentrodrigu@hotmail.fr",
    "birthdate": "14/05/1413",
    "avatar": "http://etc.usf.edu/clipart/5500/5596/columbus_14_md.gif"
  },
  {
    "id": "6",
    "username": "erozis",
    "first_name": "Emilie",
    "last_name": "Rozis",
    "email": "emilie.40@hotmail.com"
  },
  {
    "first_name": "Hubert",
    "username": "hmoncenis",
    "id": "1",
    "last_name": "Montcenis"
  },
  {
    "id": "7",
    "username": "vivianeglz",
    "first_name": "Viviane",
    "last_name": "Gonzalez",
    "email": "viviane.gonzalez@outlook.fr",
    "birthdate": "02/06/1995",
    "avatar": "kikou.png"
  },
  {
    "id": "3",
    "username": "antoinesinety",
    "first_name": "Antoine",
    "last_name": "de Sinety",
    "email": "fake@yopmail.fr",
    "birthdate": "08/06/1995",
    "avatar": "Image.png"
  },
  {
    "id": "4",
    "username": "tgauzin",
    "first_name": "Thomas",
    "last_name": "Gauzin",
    "email": "thomas.gauzin@gmail.com"
  }
]
