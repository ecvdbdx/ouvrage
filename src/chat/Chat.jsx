import React from 'react';
import Contacts from './Contacts.jsx';
import Conversation from './Conversation.jsx';

export default class Chat extends React.Component {
  constructor (props) {
    super(props);
    this.openConversation = this.openConversation.bind(this);
    this.state = {
      currentContact: null
    };
  }
  openConversation (contact) {
    this.setState({
      currentContact: contact
    });
  }

  render () {
    return (
      <div id="chat">
        <Contacts students={this.props.students} openConversation={this.openConversation} />
        <Conversation currentContact={this.state.currentContact}/>
      </div>
    );
  }
}
