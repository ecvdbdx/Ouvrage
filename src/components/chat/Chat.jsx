import React from 'react';
import Contacts from './Contacts.jsx';
import Conversation from './Conversation.jsx';
import ChooseMe from './ChooseMe.jsx';
import ChatApi from './ChatApi.js';

export default class Chat extends React.Component {
  constructor () {
    super();
    this.openConversation = this.openConversation.bind(this);
    this.selectMe = this.selectMe.bind(this);
    this.state = {
      currentContact: null,
      me: null,
      students: []
    };

    ChatApi.getStudents().then((students) => {
      this.setState({
        students: students,
        me: null
      });
    }).catch((err) => {
      console.log(err);
    });
  }

  openConversation (contact) {
    this.setState({
      currentContact: contact
    });
  }

  selectMe (currentUser) {
    this.setState({
      me: currentUser
    });
  }

  render () {
    const currentContactActif = this.state.currentContact !== null;
    const isMe = this.state.me === null;

    const component = isMe ?
      (
        <ChooseMe students={this.state.students} selectMe={this.selectMe} />
      ) : (
        <div>
          <Contacts students={this.state.students}
            currentContact={this.state.currentContact}
            openConversation={this.openConversation} />
          {
            currentContactActif ?
              (
                <Conversation currentContact={this.state.currentContact} me={this.state.me} />
              ) : null
          }
        </div>
      );

    return (
      <div className="chat">
        {component}
      </div>
    );
  }
}
