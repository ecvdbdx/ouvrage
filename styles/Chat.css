html, body, #root, .chat{
    height: 100%;
}

.chat{
  background-color: #fff;
  display: flex;
  font-size: 14px;
  min-width: 500px;
}

.list-conversations{
  flex: 0 0 25%;
  max-width: 420px;
  min-width: 240px;
  height: 100vh;
  overflow-y: auto;
}

.conversation{
  border-left: 1px solid rgba(0, 0, 0, .20);
  display: flex;
  flex: 3;
  flex-direction: column;
  min-width: 0;
}