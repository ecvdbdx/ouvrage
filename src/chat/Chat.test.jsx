import React from 'react';
import Chat from './Chat';
import renderer from 'react-test-renderer';
import students from './../../mocks/students.json';

it('renders correctly', () => {
  const tree = renderer
    .create(<Chat students={students} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
