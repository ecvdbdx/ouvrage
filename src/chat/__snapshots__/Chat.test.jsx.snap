// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="chat"
>
  <div
    id="list-conversations"
  >
    <h2>
      Les contacts
    </h2>
    <ul>
      <li
        onClick={[Function]}
      >
        epatrois
      </li>
      <li
        onClick={[Function]}
      >
        mdewaele
      </li>
      <li
        onClick={[Function]}
      >
        rodrigodelanoche
      </li>
      <li
        onClick={[Function]}
      >
        erozis
      </li>
      <li
        onClick={[Function]}
      >
        hmoncenis
      </li>
      <li
        onClick={[Function]}
      >
        vivianeglz
      </li>
      <li
        onClick={[Function]}
      >
        antoinesinety
      </li>
      <li
        onClick={[Function]}
      >
        tgauzin
      </li>
    </ul>
  </div>
  <div />
</div>
`;
